src/components/course/routes/ExternalCourseEnrollment.jsx

```diff
@@ -19,18 +19,22 @@ const ExternalCourseEnrollment = () => {
       courseEntitlementProductSku,
     },
     userSubsidyApplicableToCourse,
+    hasSuccessfulRedemption,
   } = useContext(CourseContext);
   const courseMetadata = useMinimalCourseMetadata();
 
   const {
     failureReason,
     failureMessage,
   } = useExternalEnrollmentFailureReason();
-
   const handleCheckoutSuccess = () => {
     history.push('enroll/complete');
   };
 
+  if (hasSuccessfulRedemption) {
+    history.push('enroll/complete');
+  }
```

Comment:
nit: I might recommend putting this logic into a `useEffect` so that we don't call `history.push` on every component re-render when `hasSuccessfulRedemption` is truthy (i.e., only the first render `hasSuccessfulRedemption` is truthy).